### PR TITLE
refactor(ekf_localizer): remove current_ekf_pose and so on

### PR DESCRIPTION
**Pull request by kminoda**
## Description

Remove some member variables

## Tests performed

Checked that the sample-rosbag tutorial works.

## Effects on system behavior

No effect expected

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

- Refactor: Updated the `KalmanFilter` class to improve const-correctness by modifying function signatures to accept constant references to `Eigen::MatrixXd` objects.
- Refactor: Enhanced modularity and encapsulation in the `EKFLocalizer` class by removing unnecessary member variables and introducing dedicated functions for retrieving the current pose and twist.

### CODE REVIEW

File: localization/ekf_localizer/src/ekf_localizer.cpp [hunk 1]
```diff
@@ -246,27 +257,32 @@ void EKFLocalizer::timerCallback()
...
+  current_ekf_pose.header.frame_id = params_.pose_frame_id;
+  current_ekf_pose.header.stamp = this->now();
+  current_ekf_pose.pose.position = tier4_autoware_utils::createPoint(x, y, z);
+  if (get_biased_yaw) {
+    current_ekf_pose.pose.orientation =
+      tier4_autoware_utils::createQuaternionFromRPY(roll, pitch, biased_yaw);
+  } else {
+    current_ekf_pose.pose.orientation =
+      tier4_autoware_utils::createQuaternionFromRPY(roll, pitch, yaw);
+  }
```
Comment (Motsu-san) [inline localization/ekf_localizer/src/ekf_localizer.cpp:271]: This seems to change `current_ekf_pose.pose.orientation` and to be a bugfix. Not matching to this PR's purpose, so could you separate from this pull request?

Comment (kminoda) [reply localization/ekf_localizer/src/ekf_localizer.cpp:271]: Sorry I didn't understand the point :pray: 
The part you've mentioned is now in a new function `getCurrentEKFPose` with `get_biased_yaw` flag. The users can select which pose to retrieve with the flag. And the function is now used in l. 239-240.

**Review by yukkysaito — APPROVED** [review]
LGTM

File: localization/ekf_localizer/src/ekf_localizer.cpp (hunk 1 above)
Comment (Motsu-san) [reply localization/ekf_localizer/src/ekf_localizer.cpp:271]: That makes sense. Thank you.

**Review by Motsu-san — APPROVED** [review]
[LGTMeow](https://lgtmeow.com)

**Pull request merged**